### ISSUE TITLE
Further entrypoints to linelist data class

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#414]
In PR #390 we added an initial single entry point to `epidist_linelist` requiring the following input columns:

```r
as_epidist_linelist <- function(
  data, pdate_lwr = NULL, pdate_upr = NULL, sdate_lwr = NULL, sdate_upr = NULL, obs_date = NULL
) {
```

There might be other entry points we would want to add. For example, we also needed a temporary entry point for those that already have time data available:

```r
as_epidist_linelist_time <- function(data) {
  class(data) <- c("epidist_linelist", class(data))
  epidist_validate_data(data)
  return(data)
}
```

As part of this issue I think we should scope out and implement the entry points to data being in the `epidist_linelist` class.

**Comment by athowes** [comment]
@seabbs writes that:

> My preference would be this is part of the simulation workflow and likely works via adding dates to simulated times but as above I think it needs some thought

Here "simulation workflow" refers to a possible section of the package for working with simulated data.

This supposes that a user would usually/only have time data if it were simulated. It's not clear to me that this is true.

**Comment by athowes** [comment]
<img width="304" alt="image" src="https://github.com/user-attachments/assets/305b7da2-8bc5-4bfd-9b61-9e6830abcdb7">

**Comment by seabbs** [comment]
I think this was closed by #441
